packages/primitives/tooltip/src/tooltip-root.directive.ts

```diff
@@ -0,0 +1,285 @@
+    readonly tooltipTriggerElementRef = contentChild.required(RdxTooltipTriggerDirective, { read: ElementRef });
+
+    private openTimer = 0;
+    private skipDelayTimer = 0;
+    private overlayRef?: OverlayRef;
+    private instance?: ViewRef;
+    private portal: TemplatePortal<unknown>;
+    private isControlledExternally = false;
+
+    ngOnInit(): void {
+        if (this.defaultOpen()) {
+            this.handleOpen();
+        }
+
+        this.isControlledExternally = this.open() !== undefined;
+    }
+
+    onTriggerEnter(): void {
+        if (this.isControlledExternally) {
+            return;
+        }
+
+        if (this.isOpenDelayed()) {
+            this.handleDelayedOpen();
+        } else {
+            this.handleOpen();
+        }
+    }
+
+    onTriggerLeave(): void {
+        window.clearTimeout(this.openTimer);
+        this.handleClose();
+    }
+
+    onOpen(): void {
+        window.clearTimeout(this.skipDelayTimer);
+        this.isOpenDelayed.set(false);
+    }
+
+    onClose(): void {
+        window.clearTimeout(this.skipDelayTimer);
+
+        this.skipDelayTimer = window.setTimeout(() => {
+            this.isOpenDelayed.set(true);
+        }, this.tooltipConfig.skipDelayDuration);
+    }
+
+    handleOpen(): void {
+        if (this.isControlledExternally) {
+            return;
+        }
+
+        this.wasOpenDelayed.set(false);
+        this.setOpen(true);
+    }
+
+    handleClose(): void {
+        if (this.isControlledExternally) {
+            return;
+        }
+
+        window.clearTimeout(this.openTimer);
+        this.setOpen(false);
+    }
+
+    private handleOverlayKeydown(): void {
+        if (!this.overlayRef) {
+            return;
+        }
+
+        this.overlayRef
+            .keydownEvents()
+            .pipe(
+                filter((event) => event.key === 'Escape'),
+                takeUntilDestroyed(this.destroyRef)
+            )
+            .subscribe((event) => {
+                this.tooltipContentDirective().onEscapeKeyDown.emit(event);
+
+                if (!event.defaultPrevented) {
+                    this.handleClose();
+                }
+            });
+    }
+
+    private handlePointerDownOutside(): void {
+        if (!this.overlayRef) {
+            return;
+        }
+
+        this.overlayRef
+            .outsidePointerEvents()
+            .pipe(takeUntilDestroyed(this.destroyRef))
+            .subscribe((event) => this.tooltipContentDirective().onPointerDownOutside.emit(event));
+    }
+
+    private handleDelayedOpen(): void {
+        window.clearTimeout(this.openTimer);
+
+        this.openTimer = window.setTimeout(() => {
+            this.wasOpenDelayed.set(true);
+            this.setOpen(true);
+        }, this.delayDuration());
+    }
+
+    private setOpen(open = false): void {
+        if (open) {
+            this.onOpen();
+
+            document.dispatchEvent(new CustomEvent('tooltip.open'));
```

Comment:
use the same thing 

```
if (isPlatformBrowser(this.platformId)) {

```

Reply:
you can inject [DOCUMENT](https://angular.dev/api/common/DOCUMENT)

Reply:
fixed